src/http.ts

```diff
@@ -363,10 +357,8 @@ export async function getAnalysis(
 ): Promise<Result<GetAnalysisResponseDto, GetAnalysisErrorCodes>> {
   const config: Payload = {
     headers: {
-      ...prepareTokenHeaders(options.sessionToken),
-      source: options.source,
-      ...(options.requestId && { 'snyk-request-id': options.requestId }),
-      ...(options.org && { 'snyk-org-name': options.org }),
+      ...prepareHeaders(options),
+      ...(options.analysisContext?.org?.name && { 'snyk-org-name': options.analysisContext.org.name }),
```

Comment:
q: Why the duplication here? We are pulling the org name from `options.org` but then also overriding it with the `analysisContext`? Why are we doing this here but not for the `createBundle/extendBundle` calls?

Reply:
this is a mess, i agree, but i'm trying to minimise my refactoring, while still achieve goals.
So, the answer is createBundle/extendBundle does not accept analysisContext.
Maybe it would make sense to add it there, but it's a call I leave to you folks later on.

Reply:
let me do a bit more refactoring, as this is a point of friction now.
I'll make org available only for bundle requests. to remove duplication

Reply:
I'm still not sure I understand - under what circumstances would the org name not be passed as `options.org` but included in `options.analysisContext.org.name`?

Reply:
pretty much this is only way it works now. So, now externally only analysisContext is used for "analyzeFolders" function call. noone uses org.
Then, org is only injected internally for bundle creating: https://github.com/snyk/code-client/blob/master/src/bundles.ts#L256
I did not touch this part in this PR.

Reply:
let me clean this code and i'll ask you to re-review a cleaner version

Reply:
@strassl-snyk pls re-review